The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5488051569/jobs/10000354418) [(plain)](https://github.com/rust-lang/rust/commit/419f0c7335850a4f114627a23ac1370a5019a510/checks/14862919045/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
##[group]Run git config --global core.autocrlf false
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
  lfs: false
  submodules: false
  set-safe-directory: true
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/should-skip-this.sh
src/ci/scripts/should-skip-this.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/verify-channel.sh
src/ci/scripts/verify-channel.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/collect-cpu-stats.sh
src/ci/scripts/collect-cpu-stats.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/install-sccache.sh
src/ci/scripts/install-sccache.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/select-xcode.sh
src/ci/scripts/select-xcode.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-clang.sh
src/ci/scripts/install-clang.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-wix.sh
src/ci/scripts/install-wix.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/disable-git-crlf-conversion.sh
src/ci/scripts/disable-git-crlf-conversion.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/install-msys2.sh
src/ci/scripts/install-msys2.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-mingw.sh
src/ci/scripts/install-mingw.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-ninja.sh
src/ci/scripts/install-ninja.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/enable-docker-ipv6.sh
src/ci/scripts/enable-docker-ipv6.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/disable-git-crlf-conversion.sh
src/ci/scripts/disable-git-crlf-conversion.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/verify-line-endings.sh
src/ci/scripts/verify-line-endings.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/verify-backported-commits.sh
src/ci/scripts/verify-backported-commits.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/verify-stable-version-number.sh
src/ci/scripts/verify-stable-version-number.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/run-build-from-ci.sh
src/ci/scripts/run-build-from-ci.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 419f0c7335850a4f114627a23ac1370a5019a510
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
failures:

---- [ui] tests/ui/impl-trait/issue-51185.rs#next stdout ----

error in revision `next`: test compilation failed although it shouldn't!
status: exit status: 101
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/impl-trait/issue-51185.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--cfg" "next" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=/checkout/tests/ui=fake-test-src-base" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/impl-trait/issue-51185.next/a" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/impl-trait/issue-51185.next/auxiliary" "-Ztrait-solver=next"
stdout: none
--- stderr -------------------------------
##[error]error: internal compiler error: compiler/rustc_codegen_llvm/src/context.rs:1010:21: `fn_abi_of_instance(foo, [])` failed: Layout(Unknown(impl std::convert::Into<for<'a> fn(&'a ())>))
  --> fake-test-src-base/impl-trait/issue-51185.rs:4:1
   |
LL | fn foo() -> impl Into<for<'a> fn(&'a ())> {

thread 'rustc' panicked at 'Box<dyn Any>', /checkout/compiler/rustc_errors/src/lib.rs:995:33
stack backtrace:
   0:     0x7fa968260cf1 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h4398e7b0ee82c655
---
   4:     0x7fa968263c9a - std::panicking::default_hook::{{closure}}::he49cb8923d8e65a5
   5:     0x7fa968263989 - std::panicking::default_hook::h87d0bdcf3098770d
   6:     0x7fa968f6509b - rustc_driver_impl[fcf16722b87e8ff9]::install_ice_hook::{closure#0}
   7:     0x7fa9682643de - std::panicking::rust_panic_with_hook::hfbfd1903705b50a5
   8:     0x7fa9692b62b3 - std[13a1b52b847f1eb5]::panicking::begin_panic::<rustc_errors[f6bb5c9606afe568]::ExplicitBug>::{closure#0}
   9:     0x7fa9692a9476 - std[13a1b52b847f1eb5]::sys_common::backtrace::__rust_end_short_backtrace::<std[13a1b52b847f1eb5]::panicking::begin_panic<rustc_errors[f6bb5c9606afe568]::ExplicitBug>::{closure#0}, !>
  10:     0x7fa968b4d7d6 - std[13a1b52b847f1eb5]::panicking::begin_panic::<rustc_errors[f6bb5c9606afe568]::ExplicitBug>
  11:     0x7fa969343b05 - <rustc_errors[f6bb5c9606afe568]::HandlerInner>::span_bug::<rustc_span[ac1a563e6c3af051]::span_encoding::Span, alloc[735b5acdc9ded0c6]::string::String>
  12:     0x7fa9693438e9 - <rustc_errors[f6bb5c9606afe568]::Handler>::span_bug::<rustc_span[ac1a563e6c3af051]::span_encoding::Span, alloc[735b5acdc9ded0c6]::string::String>
  13:     0x7fa9692db438 - rustc_middle[3b8e2c597d0c5895]::util::bug::opt_span_bug_fmt::<rustc_span[ac1a563e6c3af051]::span_encoding::Span>::{closure#0}
  14:     0x7fa9692db48c - rustc_middle[3b8e2c597d0c5895]::ty::context::tls::with_opt::<rustc_middle[3b8e2c597d0c5895]::util::bug::opt_span_bug_fmt<rustc_span[ac1a563e6c3af051]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  15:     0x7fa9692da6a4 - rustc_middle[3b8e2c597d0c5895]::ty::context::tls::with_context_opt::<rustc_middle[3b8e2c597d0c5895]::ty::context::tls::with_opt<rustc_middle[3b8e2c597d0c5895]::util::bug::opt_span_bug_fmt<rustc_span[ac1a563e6c3af051]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  16:     0x7fa968b4d7c7 - rustc_middle[3b8e2c597d0c5895]::util::bug::span_bug_fmt::<rustc_span[ac1a563e6c3af051]::span_encoding::Span>
  17:     0x7fa969231971 - <rustc_codegen_llvm[8eb4ba12d90aeabf]::context::CodegenCx as rustc_middle[3b8e2c597d0c5895]::ty::layout::FnAbiOfHelpers>::handle_fn_abi_err
  18:     0x7fa96922283e - <rustc_codegen_llvm[8eb4ba12d90aeabf]::context::CodegenCx as rustc_middle[3b8e2c597d0c5895]::ty::layout::FnAbiOf>::fn_abi_of_instance::{closure#0}
  19:     0x7fa969234a3a - <rustc_codegen_llvm[8eb4ba12d90aeabf]::context::CodegenCx as rustc_codegen_ssa[71c14ac395324933]::traits::declare::PreDefineMethods>::predefine_fn
  20:     0x7fa96930710c - rustc_codegen_llvm[8eb4ba12d90aeabf]::base::compile_codegen_unit::module_codegen
  21:     0x7fa969305ea4 - rustc_codegen_llvm[8eb4ba12d90aeabf]::base::compile_codegen_unit
  22:     0x7fa9691f9234 - <rustc_codegen_llvm[8eb4ba12d90aeabf]::LlvmCodegenBackend as rustc_codegen_ssa[71c14ac395324933]::traits::backend::CodegenBackend>::codegen_crate
  23:     0x7fa96914808f - <rustc_session[f8c9107426fa889b]::session::Session>::time::<alloc[735b5acdc9ded0c6]::boxed::Box<dyn core[fb902e68b063d2d1]::any::Any>, rustc_interface[45c2e7d8043f0f0]::passes::start_codegen::{closure#0}>
  24:     0x7fa969110348 - rustc_interface[45c2e7d8043f0f0]::passes::start_codegen
  25:     0x7fa96917e2fd - <rustc_middle[3b8e2c597d0c5895]::ty::context::GlobalCtxt>::enter::<<rustc_interface[45c2e7d8043f0f0]::queries::Queries>::ongoing_codegen::{closure#0}, core[fb902e68b063d2d1]::result::Result<alloc[735b5acdc9ded0c6]::boxed::Box<dyn core[fb902e68b063d2d1]::any::Any>, rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>>
  26:     0x7fa969146bf6 - <rustc_interface[45c2e7d8043f0f0]::queries::Queries>::ongoing_codegen
  27:     0x7fa968fcb812 - <rustc_interface[45c2e7d8043f0f0]::interface::Compiler>::enter::<rustc_driver_impl[fcf16722b87e8ff9]::run_compiler::{closure#1}::{closure#2}, core[fb902e68b063d2d1]::result::Result<core[fb902e68b063d2d1]::option::Option<rustc_interface[45c2e7d8043f0f0]::queries::Linker>, rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>>
  28:     0x7fa968f75559 - rustc_span[ac1a563e6c3af051]::set_source_map::<core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>, rustc_interface[45c2e7d8043f0f0]::interface::run_compiler<core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>, rustc_driver_impl[fcf16722b87e8ff9]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  29:     0x7fa968f8e919 - std[13a1b52b847f1eb5]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[45c2e7d8043f0f0]::util::run_in_thread_pool_with_globals<rustc_interface[45c2e7d8043f0f0]::interface::run_compiler<core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>, rustc_driver_impl[fcf16722b87e8ff9]::run_compiler::{closure#1}>::{closure#0}, core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>>
  30:     0x7fa968f8a6a8 - std[13a1b52b847f1eb5]::panicking::try::<core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>, core[fb902e68b063d2d1]::panic::unwind_safe::AssertUnwindSafe<<std[13a1b52b847f1eb5]::thread::Builder>::spawn_unchecked_<rustc_interface[45c2e7d8043f0f0]::util::run_in_thread_pool_with_globals<rustc_interface[45c2e7d8043f0f0]::interface::run_compiler<core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>, rustc_driver_impl[fcf16722b87e8ff9]::run_compiler::{closure#1}>::{closure#0}, core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
  31:     0x7fa968fe72b1 - <<std[13a1b52b847f1eb5]::thread::Builder>::spawn_unchecked_<rustc_interface[45c2e7d8043f0f0]::util::run_in_thread_pool_with_globals<rustc_interface[45c2e7d8043f0f0]::interface::run_compiler<core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>, rustc_driver_impl[fcf16722b87e8ff9]::run_compiler::{closure#1}>::{closure#0}, core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[fb902e68b063d2d1]::result::Result<(), rustc_span[ac1a563e6c3af051]::ErrorGuaranteed>>::{closure#1} as core[fb902e68b063d2d1]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  33:     0x7fa968007b43 - <unknown>
  34:     0x7fa968099a00 - <unknown>
  35:                0x0 - <unknown>


note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.72.0-nightly (8405cacd9 2023-07-07) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z threads=1 -Z simulate-remapped-rust-src-base=/rustc/FAKE_PREFIX -Z translate-remapped-path-to-local-path=no -C codegen-units=1 -Z ui-testing -Z deduplicate-diagnostics=no -C strip=debuginfo -C prefer-dynamic -C rpath -C debuginfo=0 -Z trait-solver=next
query stack during panic:
end of query stack
error: aborting due to previous error
------------------------------------------
```

</details>

--- a/tests/ui/traits/new-solver/alias-bound-preference.rs
+++ b/tests/ui/traits/new-solver/alias-bound-preference.rs
@@ -1,0 +1,39 @@
+// revisions: old next
+//[next] compile-flags: -Ztrait-solver=next
+// run-pass
+
+// A test for https://github.com/rust-lang/trait-system-refactor-initiative/issues/45.
+
+trait Trait {
+    type Assoc: Into<u32>;
+}
+impl<T: Into<u32>> Trait for T {
+    type Assoc = T;
+}
+fn prefer_alias_bound_projection<T: Trait>(x: T::Assoc) {
+    // There are two possible types for `x`:
+    // - `u32` by using the "alias bound" of `<T as Trait>::Assoc`
+    // - `<T as Trait>::Assoc`, i.e. `u16`, by using `impl<T> From<T> for T`
+    //
+    // We infer the type of `x` to be `u32` here as it is highly likely
+    // that this is expected by the user.
+    let x = x.into();
+    assert_eq!(std::mem::size_of_val(&x), 4);
+}
+
+fn impl_trait() -> impl Into<u32> {
+    0u16
+}
+
+fn main() {
+    // There are two possible types for `x`:
+    // - `u32` by using the "alias bound" of `impl Into<u32>`
+    // - `impl Into<u32>`, i.e. `u16`, by using `impl<T> From<T> for T`
+    //
+    // We infer the type of `x` to be `u32` here as it is highly likely
+    // that this is expected by the user.
+    let x = impl_trait().into();
+    assert_eq!(std::mem::size_of_val(&x), 4);
+
+    prefer_alias_bound_projection::<u16>(1);
+}

